packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -160,52 +160,89 @@ - (UIImage *)extractIconFromData:(NSArray *)iconData
-                                       reason:@"Unable to interpret bytes as a valid image."
-                                     userInfo:nil];
+        // Update proper scale information for image object.
+        image = [self scaleImage:image withScale:screenScale];
+        // Create resized image.
+        CGSize size_param = [FLTGoogleMapJSONConversions sizeFromArray:iconData[4]];
+        CGSize size = [self scaleSizeToInt:size_param withFactor:screenScale];
+        image = [self scaleImage:image toSize:size];
       }
-    } else {
-      NSString *error = [NSString
-          stringWithFormat:@"fromBytes should have exactly one argument, the bytes. Got: %lu",
-                           (unsigned long)iconData.count];
-      NSException *exception = [NSException exceptionWithName:@"InvalidByteDescriptor"
-                                                       reason:error
-                                                     userInfo:nil];
-      @throw exception;
+    }
+
+  } else if ([iconData[0] isEqualToString:@"bytes"]) {
+    NSAssert((iconData.count > 2), @"'bytes' should have at least 3 parameters. Got: %lu",
+             (unsigned long)iconData.count);
+    @try {
+      FlutterStandardTypedData *byteData = iconData[1];
+      if ([iconData[2] isEqualToString:@"auto"]) {
+        NSAssert((iconData.count == 4 || iconData.count == 5),
+
+                 @"'bytes' with auto setting should have exactly 4 or 5 arguments, Got: %lu",
+                 (unsigned long)iconData.count);
+        if (iconData.count == 4) {
+          CGFloat imageScale = [iconData[3] doubleValue];
+          // Scale parameter given, use it as scale factor
+          image = [UIImage imageWithData:[byteData data] scale:imageScale];
+        } else if (iconData.count == 5) {
+          CGFloat screenScale = [[UIScreen mainScreen] scale];
+          // Size parameter is given, scale image to exact size.
+          // Update proper scale information for image object.
+          image = [UIImage imageWithData:[byteData data] scale:screenScale];
+          // Create resized image.
+          CGSize size_param = [FLTGoogleMapJSONConversions sizeFromArray:iconData[4]];
+          CGSize size = [self scaleSizeToInt:size_param withFactor:screenScale];
+          image = [self scaleImage:image toSize:size];
+        }
+      } else {
+        // No scaling, load image from bytes without scale parameter.
+        image = [UIImage imageWithData:[byteData data]];
+      }
+    } @catch (NSException *exception) {
+      @throw [NSException exceptionWithName:@"InvalidByteDescriptor"
+                                     reason:@"Unable to interpret bytes as a valid image."
+                                   userInfo:nil];
     }
   }
 
   return image;
 }
 
+// Used by deprecated fromBytes functionality.
+// Can be removed when deprecated image bitmap types are removed from platform interface.
```

Comment:
Can you link to the exact type signature here? It is hard to find out what `deprecated image bitmap types` are when we want to clean up the code later. It would be nice to create a Github issue to track this and add a TODO here.

Reply:
Update the docs for this method to describe the deprecation better